odoo/service/server.py

```diff
@@ -367,15 +367,17 @@ def start(self):
         import gevent
         from gevent.wsgi import WSGIServer
 
-        # Set process memory limit as an extra safeguard
-        _, hard = resource.getrlimit(resource.RLIMIT_AS)
-        resource.setrlimit(resource.RLIMIT_AS, (config['limit_memory_hard'], hard))
 
         if os.name == 'posix':
+            # Set process memory limit as an extra safeguard
+            _, hard = resource.getrlimit(resource.RLIMIT_AS)
+            resource.setrlimit(resource.RLIMIT_AS, (config['limit_memory_hard'], hard))
             signal.signal(signal.SIGQUIT, dumpstacks)
             signal.signal(signal.SIGUSR1, log_ormcache_stats)
-
-        gevent.spawn(self.watchdog)
+            gevent.spawn(self.watchdog)
+        else:
```

Comment:
The `else` block seems useless here, was it supposed to achieve anything? `spawn()` without argument will start a greenlet with an empty `run()` method, IIRC.

Reply:
If it needed, if you try to launch `gevent.spawn(self.watchdog)` you will find the same problem that with the `resource` module on non-posix systems. In that way we can start a greenlet.

Reply:
Yes, you can't call `gevent.spawn(self.watchdog)` on non-posix systems. But why do you call `gevent.spawn()` with no argument? It starts an empty greenlet thread. What's the point of that?

Reply:
@odony , you are right, the 'else' block is completely useless. Sorry for the confusion. Shall I modify the PR?

Reply:
I think the best option is to remove it.

Reply:
Sure, just remove it in your branch and push it as an extra commit, this will update the PR.
You can also amend and force-push your initial commit on the branch, same thing, we'll squash it when merging anyway.
Thanks!

Reply:
Done